CRM/MembershipExtras/Job/OfflineAutoRenewal/MultipleInstallmentPlan.php

```diff
@@ -131,6 +132,63 @@ private function createRecurringContribution() {
     $this->newRecurringContributionID = $newRecurringContribution['id'];
   }
 
+  /**
+   * Calculates the new period's start date from the largest membership end date
+   * of the previous period.
+   *
+   * @return string
+   *   The new period's start date.
+   * @throws \Exception
+   */
+  private function calculateNewPeriodStartDate() {
+    $latestDate = NULL;
+    $currentPeriodLines = $this->getRecurringContributionLineItemsToBeRenewed($this->currentRecurContributionID);
+    foreach ($currentPeriodLines as $lineItem) {
```

Comment:
I know it is generally better to use the API, but I think here it probably better if we optimize it by fetching and sorting the results with SQL and joins instead.